src/toil/test/utils/utilsTest.py

```diff
@@ -183,6 +186,128 @@ def testAWSProvisionerUtils(self):
                 shutil.rmtree(tmpDir)
             except NameError:
                 pass
+    def _createSortOptions(self, jobstoreLoc, fails=False):
+        options = Job.Runner.getDefaultOptions(jobstoreLoc)
+        options.badWorker = fails
+        options.clean = 'never'  # Will delete jobstore after the testing is done.
+        options.logLevel = 'info'
+        options.overwriteOutput = True
+        options.outputFile = self.outputFile
+        options.fileToSort = None
+        options.downCheckpoints = False
+        options.N = self.N
+        options.debugWorker = False
+        options.sleepDuration = 15
+        options.numLines = 50
+        options.lineLength = 20  # Ensures that there is enough time to get a running response from before it finshes.
+        return options
+
+    def testGetPIDStatus(self):
+        """Test that ToilStatus.getPIDStatus() behaves as expected."""
+        jobstoreName = 'pidStatusTest'
+        jobstoreLoc = os.path.join(os.getcwd(), jobstoreName)
+
+        cmd = ['python', '-m', 'toil.test.sort.sort', 'file:' + jobstoreName, '--clean', 'never']
+        wf = Popen(cmd)
+        time.sleep(2)  # Need to let jobstore be created before checking its contents.
+        self.assertEqual(ToilStatus.getPIDStatus(jobstoreLoc),'RUNNING')
+        wf.wait()
+        self.assertEqual(ToilStatus.getPIDStatus(jobstoreLoc), 'COMPLETED')
+        os.remove(os.path.join(jobstoreLoc,'pid.log'))
+        self.assertEqual(ToilStatus.getPIDStatus(jobstoreLoc), 'QUEUED')
+        shutil.rmtree(jobstoreLoc)
+
+    def testGetStatusFailedToilWF(self):
+        """
+        Test that ToilStatus.getStatus() behaves as expected with a failing Toil workflow.
+
+        Run a Toil workflow, with the badWorker option set to force a failure, in a subprocess. While this workflow
```

Comment:
Line 222 provides a sufficient overview of the method and is good as the docstring. Lines 224-228 should be incorporated into the source code inline as comments to explain what is happening